jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/BlockingTest.java

```diff
@@ -13,131 +13,102 @@
 
 package org.eclipse.jetty.util;
 
+import java.io.Closeable;
 import java.io.IOException;
 import java.io.InterruptedIOException;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
 import org.slf4j.LoggerFactory;
 
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.greaterThan;
-import static org.hamcrest.Matchers.lessThan;
+import static org.awaitility.Awaitility.await;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
+@Timeout(value = 10)
 public class BlockingTest
 {
     final Blocker.Shared _shared = new Blocker.Shared();
 
+    Thread main;
+
+    @BeforeEach
+    public void setUp()
+    {
+        main = Thread.currentThread();
+    }
+
     @Test
     public void testRunBlock() throws Exception
     {
-        long start;
         try (Blocker.Runnable runnable = Blocker.runnable())
         {
             runnable.run();
-            start = NanoTime.now();
             runnable.block();
         }
-        assertThat(NanoTime.millisSince(start), lessThan(500L));
     }
```

Comment:
Is there really no assertion on this specific test?
Seems like all we are doing is making sure that no exception is thrown.
Is this supposed to be testing `runnable.block()` or `runnable.run()` ?

Reply:
This test asserts that calling `run()` on the runnable before calling `block()` on it won't block. There's a 10s timeout annotation on the class that ensures this happens.